### PR TITLE
Fixed#2684: use the correct stacklevel for deprecation warnings

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

File: _pytest/config.py [hunk 1]
```diff
@@ -636,7 +636,7 @@ def __init__(self, *names, **attrs):
                 'pytest now uses argparse. "%default" should be'
                 ' changed to "%(default)s" ',
                 DeprecationWarning,
-                stacklevel=3)
```
Comment (RonnyPfannschmidt) [inline _pytest/config.py]: this change looks incorrect, the argument type is used indirectly and the higher level intended to trigger that warning for the method call side

File: _pytest/python.py [hunk 2]
```diff
@@ -919,7 +919,7 @@ def _idval(val, argname, idx, idfn, config=None):
             import warnings
             msg = "Raised while trying to determine id of parameter %s at position %d." % (argname, idx)
             msg += '\nUpdate your code as this will raise an error in pytest-4.0.'
-            warnings.warn(msg, DeprecationWarning)
+            warnings.warn(msg, DeprecationWarning, stacklevel=2)
```
Comment (RonnyPfannschmidt) [inline _pytest/python.py]: for this one stacklevel is actually not applicable

File: _pytest/mark.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def extract_from(cls, parameterset, legacy_force_tuple=False):
             argval = argval,
 
         if newmarks:
-            warnings.warn(MARK_PARAMETERSET_UNPACKING)
+            warnings.warn(MARK_PARAMETERSET_UNPACKING, stacklevel=2)
```
Comment (RonnyPfannschmidt) [inline _pytest/mark.py]: in this case stacklevel is also not usable, its a case where its probably more sensible to warn explicit and pass the test around

**Comment by RonnyPfannschmidt** [comment]
i should have been more clear with what i meant by review, because simply setting `stacklevel=2` everywhere is certainly incorrect

**Comment by nicoddemus** [comment]
I agree, unfortunately a few of those will need to use [`warn_explicit`](https://docs.python.org/3/library/warnings.html?highlight=warn_explicit#warnings.warn_explicit) because the origin of the warning is far away from the code that actually generates the warning.

**Comment by srinivasreddy** [comment]
I am closing this.

**Comment by nicoddemus** [comment]
Thanks anyway for the effort @srinivasreddy, we appreciate it! :+1:

**Pull request closed without merging**